File: FishBuy.Web/ClientApp/src/app/model/order.ts
import { OrderStatusEnum } from "./components/OrderStatusEnum";
import { OrderItem } from "./orderItem";

export class Order {
  public id: number;

  public userId: number;

  public orderDate: Date;
  public deliveryDate: Date;
  public status: OrderStatusEnum;

  public addressNumber: number;
  public address: string;
  public city: string;
  public state: string;
  public postalCode: string;

  public paymentMethodId: number;
  public total: number;

  public orderItems: OrderItem[];

  constructor() {
    this.orderDate = new Date();
    this.orderItems = [];
  }
}

File: FishBuy.Domain/Entities/Order.cs
using FishBuy.Domain.Enums;
using FishBuy.Domain.ObjetoDeValor;
using System;
using System.Collections.Generic;

namespace FishBuy.Domain.Entities
{
    public class Order
    {
        public int Id { get; set; }

        public int UserId { get; set; }
        public virtual User User { get; set; }

        public DateTime OrderDate { get; set; }
        public DateTime DeliveryDate { get; set; }
        public StatusEnum Status { get; set; }

        public int AddressNumber { get; set; }
        public string Address { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string PostalCode { get; set; }


        public int PaymentMethodId { get; set; }
        public virtual PaymentMethod PaymentMethod { get; set; }
        public float Total { get; set; }

        public virtual ICollection<OrderItem> OrderItems { get; set; }

    }
}
